### PR TITLE
Phantom Upgrade - Adding the ability to see Playbook Run History and create Events/Containers when searching On Demand

### PR DESCRIPTION
**Pull request by penwoodjon**
## Steps to Test
1.) Install Integration and add credentials from 1pass
2.) Change the config rejectUnauthorized to false, turn off caching, and restart the Integration
3.) Search these keys and verify the responses in the overlay seem reasonable and contain Playbook Run History that looks correct (reference postman collection as needed):
```
100.100.100.100
192.185.91.42
178.128.76.6
```
4.)  Search `docs.microsoft.com` onDemand and verify the panel that appears in the overlay works and looks correct.
5.) Review code and give feedback on 
- [ ] Formatting/Readability
- [ ] Test cases
- [ ] Actual functionality being executed (e.g. Auth, requests, data transformations)

> **Note:** This Integration allows for creating Events/Containers in Phantom when searching for entities On Demand.  It is important that if you create an Event/Container in Phantom while testing, you to delete the newly created Event/Container after you are done.

### CODE REVIEW

**Comment by radar989** [comment]
@penwoodjon need to update the package.json file got 3 critical vulns when installing the integration from scratch.

**Comment by radar989** [comment]
need to increment the version to 3.1 as well

**Comment by radar989** [comment]
@penwoodjon Got this error when running the integration for the first time: `stack": "TypeError: Cannot read property 'length' of undefined\n    at containers.map (/app/polarity-server/integrations/phantom/containers.js:25:24`

**Comment by sarus** [comment]
@penwoodjon 

Seeing this error when running a lookup that has no results (try running a lookup on the IP `
54.2.3.2`

```
{
    "debug": {
        "message": "Cannot read property 'length' of undefined",
        "stack": "TypeError: Cannot read property 'length' of undefined\n    at containers.map (/home/ed/workspace/polarity-server/integrations/phantom/containers.js:25:24)\n    at Array.map (<anonymous>)\n    at _getContainers (/home/ed/workspace/polarity-server/integrations/phantom/containers.js:22:42)\n    at async.each (/home/ed/workspace/polarity-server/integrations/phantom/containers.js:60:16)\n    at /home/ed/workspace/polarity-server/integrations/phantom/node_modules/async/dist/async.js:473:16\n    at iteratorCallback (/home/ed/workspace/polarity-server/integrations/phantom/node_modules/async/dist/async.js:1064:13)\n    at /home/ed/workspace/polarity-server/integrations/phantom/node_modules/async/dist/async.js:969:16\n    at _getContainerSearchResults (/home/ed/workspace/polarity-server/integrations/phantom/containers.js:56:47)\n    at Request.request [as _callback] (/home/ed/workspace/polarity-server/integrations/phantom/containers.js:87:16)\n    at Request.self.callback (/home/ed/workspace/polarity-server/integrations/phantom/node_modules/request/request.js:185:22)"
    },
    "error-signature": "2924411daf72979691260bc4ce75c873"
}
```

**Comment by sarus** [comment]
Additional feedback.

1. When a playbook is run it does not update the playbook run history.  After a playbook runs we should refresh the details block for the container that has the playbook run.

2. I don't see any quick link to open the indicator in phantom.

3. Can playbook histories link to anything if you click on it? Is there a details page for the run of the playbook we could send users to?

4. If a playbook fails all you get back is a "error running playbook" message.  Would be nice to include some of the error message coming back.  For example, if you try to run the "c2_investigate_and_contain" playbook you get back the error:

```
 {
     "message": "'geolocate_ip_1' action executed for 0 parameters, 'ip_reputation_1' action executed for 0 parameters, 'hunt_ip_1' action executed for 0 parameters, 'whois_ip_1' action executed for 0 parameters, 'list_vms_1' action executed for 0 parameters, 'block_ip_1' action executed for 0 parameters",
     "name": "community/c2_investigate_and_contain(id=6) playbook_run_id:10",
     "playbook": "community/c2_investigate_and_contain",
     "playbook_run_id": 10,
     "result": [
     {
         "action": "geolocate ip",
         "action_run_id": 7,
         "app_runs": null,
         "callback":
         {
             "cb_called": "yes",
             "cb_fn_name": "join_filter_4",
             "cb_result": true
         },
         "close_time": "2020-02-18T15:51:22.158421+00:00",
         "create_time": "2020-02-18T15:51:22.153+00:00",
         "id": 7,
         "message": "phantom.act(): 'geolocate_ip_1' cannot be executed on asset 'maxmind'. Required parameter 'ip' not provided\\n",
         "name": "geolocate_ip_1",
         "responses": null,
         "status": "failed",
         "type": ""
     },
     {
         "action": "ip reputation",
         "action_run_id": 8,
         "app_runs": null,
         "close_time": "2020-02-18T15:51:22.190246+00:00",
         "create_time": "2020-02-18T15:51:22.184+00:00",
         "id": 8,
         "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
         "name": "ip_reputation_1",
         "responses": null,
         "status": "failed",
         "type": ""
     },
     {
         "action": "hunt ip",
         "action_run_id": 9,
         "app_runs": null,
         "close_time": "2020-02-18T15:51:22.219209+00:00",
         "create_time": "2020-02-18T15:51:22.216+00:00",
         "id": 9,
         "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
         "name": "hunt_ip_1",
         "responses": null,
         "status": "failed",
         "type": ""
     },
     {
         "action": "whois ip",
         "action_run_id": 10,
         "app_runs": null,
         "close_time": "2020-02-18T15:51:22.247073+00:00",
         "create_time": "2020-02-18T15:51:22.242+00:00",
         "id": 10,
         "message": "phantom.act(): 'whois_ip_1' cannot be executed on asset 'whois'. Required parameter 'ip' not provided\\n",
         "name": "whois_ip_1",
         "responses": null,
         "status": "failed",
         "type": ""
     },
     {
         "action": "list vms",
         "action_run_id": 11,
         "app_runs": null,
         "close_time": "2020-02-18T15:51:22.277491+00:00",
         "create_time": "2020-02-18T15:51:22.263+00:00",
         "id": 11,
         "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
         "name": "list_vms_1",
         "responses": null,
         "status": "failed",
         "type": ""
     },
     {
         "action": "block ip",
         "action_run_id": 12,
         "app_runs": null,
         "close_time": "2020-02-18T15:51:22.323279+00:00",
         "create_time": "2020-02-18T15:51:22.315+00:00",
         "id": 12,
         "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
         "name": "block_ip_1",
         "responses": null,
         "status": "failed",
         "type": ""
     }],
     "status": "failed"
 }
```

(note this error is in the logs, not in the interface)

We should parse this error and return back a list of actions and messages

```
{
    "errors": [
    {
        "action": "geolocate ip",
        "message": "phantom.act(): 'geolocate_ip_1' cannot be executed on asset 'maxmind'. Required parameter 'ip' not provided\\n",
    },
    {
        "action": "ip reputation",
        "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
    },
    {
        "action": "hunt ip",
        "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
    },
    {
        "action": "whois ip",
        "message": "phantom.act(): 'whois_ip_1' cannot be executed on asset 'whois'. Required parameter 'ip' not provided\\n",
    },
    {
        "action": "list vms",
        "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
    },
    {
        "action": "block ip",
        "message": "phantom.act(): Failed to get asset to execute the action. Action execution shall fail",
    }]
}
```

then display this in the overlay window.

**Comment by sarus** [comment]
Another error I ran into.  Happens now anytime I run a lookup on `192.185.91.42`

```
{
    "debug": {
        "message": "Unexpected token S in JSON at position 0",
        "stack": "SyntaxError: Unexpected token S in JSON at position 0\n    at JSON.parse (<anonymous>)\n    at body.data.map (/home/ed/workspace/polarity-server/integrations/phantom/playbooks.js:65:42)\n    at Array.map (<anonymous>)\n    at requestWithDefaults (/home/ed/workspace/polarity-server/integrations/phantom/playbooks.js:64:42)\n    at Request.request [as _callback] (/home/ed/workspace/polarity-server/integrations/phantom/error-handler.js:9:13)\n    at Request.self.callback (/home/ed/workspace/polarity-server/integrations/phantom/node_modules/request/request.js:185:22)\n    at Request.emit (events.js:182:13)\n    at Request.<anonymous> (/home/ed/workspace/polarity-server/integrations/phantom/node_modules/request/request.js:1161:10)\n    at Request.emit (events.js:182:13)\n    at IncomingMessage.<anonymous> (/home/ed/workspace/polarity-server/integrations/phantom/node_modules/request/request.js:1083:12)"
    },
    "error-signature": "320d3b689ff632b3e23a27b86b1add4c"
}
```

The problem is that we're trying to parse the `playbookRan.message` as though it's JSON but in some cases it appears it is not.  For this particular run event the message is just a string:

```
        {
          "status": "running",
          "node_guid": null,
          "update_time": "2020-02-18T17:20:38.566466Z",
          "test_mode": false,
          "log_level": 0,
          "run_data": {},
          "start_time": "2020-02-18T17:20:38.497000Z",
          "misc": {
            "scope": "new",
            "parent_playbook_run": null,
            "dbg_token": "",
            "callback": {
              "reset_password": {
                "action": "prompt",
                "cb_result": false,
                "cb_fn_name": "reset_option",
                "cb_called": "no"
              }
            }
          },
          "action_exec": [],
          "effective_user": 2,
          "owner": 2,
          "version": 1,
          "last_artifact": 2,
          "playbook": 1,
          "cancelled": null,
          "container": 2,
          "message": "Starting playbook 'community/activedirectory_reset_password (version: 1, id: 1)' on 'events'(id: 2) with playbook run id: 14, running as user 'automation'(id: 2)",
          "ip_address": "34.198.146.206",
          "id": 14
        },
```

File: README.md [hunk 1]
```diff
@@ -2,9 +2,10 @@
 
 ![image](https://img.shields.io/badge/status-beta-green.svg)
 
-Polarity's Phantom integration allows automated queries against Phantom's container and artifact database and allows a user to execute pre-defined playbooks from the Polarity overlay window.
+Polarity's Phantom integration allows automated queries against Phantom's container and artifact database, create containers from entities, and allows a user to execute pre-defined playbooks from the Polarity overlay window.
 
-![phantomint](https://user-images.githubusercontent.com/22529325/53601678-a44bf100-3b7a-11e9-9a87-c29970c127c5.gif)
+<img width="400" alt="phantomInt" src="https://user-images.githubusercontent.com/22529325/53601678-a44bf100-3b7a-11e9-9a87-c29970c127c5.gif">
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply README.md]: I'm not sure either of them will be until we work out all the kinks with the UI.  At the end of all of the changes I'll make sure to update the photo and readme.

**Pull request merged**